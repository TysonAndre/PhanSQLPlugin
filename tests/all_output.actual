array(1) {
  [0]=>
  object(PhpMyAdmin\SqlParser\Statements\SelectStatement)#15951 (16) {
    ["expr"]=>
    array(1) {
      [0]=>
      object(PhpMyAdmin\SqlParser\Components\Expression)#15953 (7) {
        ["database"]=>
        NULL
        ["table"]=>
        NULL
        ["column"]=>
        NULL
        ["expr"]=>
        string(1) "*"
        ["alias"]=>
        NULL
        ["function"]=>
        NULL
        ["subquery"]=>
        NULL
      }
    }
    ["from"]=>
    array(1) {
      [0]=>
      object(PhpMyAdmin\SqlParser\Components\Expression)#15954 (7) {
        ["database"]=>
        NULL
        ["table"]=>
        string(7) "example"
        ["column"]=>
        NULL
        ["expr"]=>
        string(7) "example"
        ["alias"]=>
        string(1) "{"
        ["function"]=>
        NULL
        ["subquery"]=>
        NULL
      }
    }
    ["partition"]=>
    NULL
    ["where"]=>
    array(3) {
      [0]=>
      object(PhpMyAdmin\SqlParser\Components\Condition)#15960 (3) {
        ["identifiers"]=>
        array(2) {
          [0]=>
          string(3) "foo"
          [1]=>
          string(3) "bar"
        }
        ["isOperator"]=>
        bool(false)
        ["expr"]=>
        string(10) "foo = :bar"
      }
      [1]=>
      object(PhpMyAdmin\SqlParser\Components\Condition)#15961 (3) {
        ["identifiers"]=>
        array(0) {
        }
        ["isOperator"]=>
        bool(true)
        ["expr"]=>
        string(3) "AND"
      }
      [2]=>
      object(PhpMyAdmin\SqlParser\Components\Condition)#15962 (3) {
        ["identifiers"]=>
        array(2) {
          [0]=>
          string(1) "x"
          [1]=>
          string(5) "HH:MM"
        }
        ["isOperator"]=>
        bool(false)
        ["expr"]=>
        string(11) "x = 'HH:MM'"
      }
    }
    ["group"]=>
    NULL
    ["having"]=>
    NULL
    ["order"]=>
    NULL
    ["limit"]=>
    NULL
    ["procedure"]=>
    NULL
    ["into"]=>
    NULL
    ["join"]=>
    NULL
    ["union"]=>
    array(0) {
    }
    ["end_options"]=>
    NULL
    ["options"]=>
    object(PhpMyAdmin\SqlParser\Components\OptionsArray)#15952 (1) {
      ["options"]=>
      array(0) {
      }
    }
    ["first"]=>
    int(0)
    ["last"]=>
    int(27)
  }
}
object(PhpMyAdmin\SqlParser\TokensList)#15933 (3) {
  ["tokens"]=>
  array(29) {
    [0]=>
    object(PhpMyAdmin\SqlParser\Token)#15931 (6) {
      ["token"]=>
      string(6) "SELECT"
      ["value"]=>
      string(6) "SELECT"
      ["keyword"]=>
      string(6) "SELECT"
      ["type"]=>
      int(1)
      ["flags"]=>
      int(3)
      ["position"]=>
      int(0)
    }
    [1]=>
    object(PhpMyAdmin\SqlParser\Token)#15928 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(6)
    }
    [2]=>
    object(PhpMyAdmin\SqlParser\Token)#15926 (6) {
      ["token"]=>
      string(1) "*"
      ["value"]=>
      string(1) "*"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(2)
      ["flags"]=>
      int(1)
      ["position"]=>
      int(7)
    }
    [3]=>
    object(PhpMyAdmin\SqlParser\Token)#15963 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(8)
    }
    [4]=>
    object(PhpMyAdmin\SqlParser\Token)#15964 (6) {
      ["token"]=>
      string(4) "FROM"
      ["value"]=>
      string(4) "FROM"
      ["keyword"]=>
      string(4) "FROM"
      ["type"]=>
      int(1)
      ["flags"]=>
      int(3)
      ["position"]=>
      int(9)
    }
    [5]=>
    object(PhpMyAdmin\SqlParser\Token)#15965 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(13)
    }
    [6]=>
    object(PhpMyAdmin\SqlParser\Token)#15966 (6) {
      ["token"]=>
      string(7) "example"
      ["value"]=>
      string(7) "example"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(14)
    }
    [7]=>
    object(PhpMyAdmin\SqlParser\Token)#15967 (6) {
      ["token"]=>
      string(1) "{"
      ["value"]=>
      string(1) "{"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(21)
    }
    [8]=>
    object(PhpMyAdmin\SqlParser\Token)#15969 (6) {
      ["token"]=>
      string(1) "{"
      ["value"]=>
      string(1) "{"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(22)
    }
    [9]=>
    object(PhpMyAdmin\SqlParser\Token)#15971 (6) {
      ["token"]=>
      string(4) "pkey"
      ["value"]=>
      string(4) "pkey"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(23)
    }
    [10]=>
    object(PhpMyAdmin\SqlParser\Token)#15972 (6) {
      ["token"]=>
      string(1) "}"
      ["value"]=>
      string(1) "}"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(27)
    }
    [11]=>
    object(PhpMyAdmin\SqlParser\Token)#15974 (6) {
      ["token"]=>
      string(1) "}"
      ["value"]=>
      string(1) "}"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(28)
    }
    [12]=>
    object(PhpMyAdmin\SqlParser\Token)#15976 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(29)
    }
    [13]=>
    object(PhpMyAdmin\SqlParser\Token)#15977 (6) {
      ["token"]=>
      string(5) "where"
      ["value"]=>
      string(5) "WHERE"
      ["keyword"]=>
      string(5) "WHERE"
      ["type"]=>
      int(1)
      ["flags"]=>
      int(3)
      ["position"]=>
      int(30)
    }
    [14]=>
    object(PhpMyAdmin\SqlParser\Token)#15978 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(35)
    }
    [15]=>
    object(PhpMyAdmin\SqlParser\Token)#15979 (6) {
      ["token"]=>
      string(3) "foo"
      ["value"]=>
      string(3) "foo"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(0)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(36)
    }
    [16]=>
    object(PhpMyAdmin\SqlParser\Token)#15980 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(39)
    }
    [17]=>
    object(PhpMyAdmin\SqlParser\Token)#15981 (6) {
      ["token"]=>
      string(1) "="
      ["value"]=>
      string(1) "="
      ["keyword"]=>
      NULL
      ["type"]=>
      int(2)
      ["flags"]=>
      int(2)
      ["position"]=>
      int(40)
    }
    [18]=>
    object(PhpMyAdmin\SqlParser\Token)#15982 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(41)
    }
    [19]=>
    object(PhpMyAdmin\SqlParser\Token)#15984 (6) {
      ["token"]=>
      string(4) ":bar"
      ["value"]=>
      string(3) "bar"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(8)
      ["flags"]=>
      int(16)
      ["position"]=>
      int(42)
    }
    [20]=>
    object(PhpMyAdmin\SqlParser\Token)#15983 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(46)
    }
    [21]=>
    object(PhpMyAdmin\SqlParser\Token)#15985 (6) {
      ["token"]=>
      string(3) "AND"
      ["value"]=>
      string(3) "AND"
      ["keyword"]=>
      string(3) "AND"
      ["type"]=>
      int(1)
      ["flags"]=>
      int(3)
      ["position"]=>
      int(47)
    }
    [22]=>
    object(PhpMyAdmin\SqlParser\Token)#15986 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(50)
    }
    [23]=>
    object(PhpMyAdmin\SqlParser\Token)#15987 (6) {
      ["token"]=>
      string(1) "x"
      ["value"]=>
      string(1) "x"
      ["keyword"]=>
      string(1) "X"
      ["type"]=>
      int(1)
      ["flags"]=>
      int(33)
      ["position"]=>
      int(51)
    }
    [24]=>
    object(PhpMyAdmin\SqlParser\Token)#15988 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(52)
    }
    [25]=>
    object(PhpMyAdmin\SqlParser\Token)#15989 (6) {
      ["token"]=>
      string(1) "="
      ["value"]=>
      string(1) "="
      ["keyword"]=>
      NULL
      ["type"]=>
      int(2)
      ["flags"]=>
      int(2)
      ["position"]=>
      int(53)
    }
    [26]=>
    object(PhpMyAdmin\SqlParser\Token)#15990 (6) {
      ["token"]=>
      string(1) " "
      ["value"]=>
      string(1) " "
      ["keyword"]=>
      NULL
      ["type"]=>
      int(3)
      ["flags"]=>
      int(0)
      ["position"]=>
      int(54)
    }
    [27]=>
    object(PhpMyAdmin\SqlParser\Token)#15991 (6) {
      ["token"]=>
      string(7) "'HH:MM'"
      ["value"]=>
      string(5) "HH:MM"
      ["keyword"]=>
      NULL
      ["type"]=>
      int(7)
      ["flags"]=>
      int(1)
      ["position"]=>
      int(55)
    }
    [28]=>
    object(PhpMyAdmin\SqlParser\Token)#15992 (6) {
      ["token"]=>
      NULL
      ["value"]=>
      NULL
      ["keyword"]=>
      NULL
      ["type"]=>
      int(9)
      ["flags"]=>
      int(0)
      ["position"]=>
      NULL
    }
  }
  ["count"]=>
  int(29)
  ["idx"]=>
  int(0)
}
src/001_exec_bind.php:3 PhanPluginOraMissingBindVar Missing bind var :bar. Actual bind vars: (:baz)
src/001_exec_bind.php:3 PhanPluginOraUnexpectedBindVar Unexpected bind var :baz. Expected bind vars: (:bar)
