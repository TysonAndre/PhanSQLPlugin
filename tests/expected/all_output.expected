src/001_exec_bind.php:3 PhanPluginOraMissingBindVar Missing bind var :bar. Actual bind vars: (:baz)
src/001_exec_bind.php:3 PhanPluginOraUnexpectedBindVar Unexpected bind var :baz. Expected bind vars: (:bar)
